compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -663,7 +663,15 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         }
         let imm = self.read_immediate_raw(op)?.right().unwrap();
         if matches!(*imm, Immediate::Uninit) {
-            throw_ub!(InvalidUninitBytes(None));
+            throw_ub!(InvalidUninitBytes(match op.to_op(self)?.as_mplace_or_imm() {
+                Left(mplace) => mplace.ptr().provenance.and_then(|prov| {
+                    let start = mplace.ptr().into_parts().1;
```

Comment:
This is not correct -- depending on which type of provenance this is, `start` will be either relative to the allocation, or absolute. `into_parts` has a [doc comment](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/interpret/struct.Pointer.html#method.into_parts) warning about this. :) Maybe we should rename it `into_raw_parts` or so to make it more clear that this API is somewhat dicey.